blockchain/txvalidator.go

```diff
@@ -1214,60 +1305,84 @@ func (b *BlockChain) checkRegisterCRTransaction(txn *Transaction) error {
 		// check code and signature
 		if err := checkStandardSignature(program.Program{
 			Code:      info.Code,
-			Parameter: parameter,
+			Parameter: info.Signature,
 		}, signedBuf.Bytes()); err != nil {
 			return err
 		}
-
-		// check DID
-		publicKey := info.Code[1 : len(info.Code)-1]
-		hash, err := contract.PublicKeyToDepositProgramHash(publicKey)
-		if err != nil {
-			return err
-		}
-		if !info.DID.IsEqual(*hash) {
-			return errors.New("the DID needs to be calculated from standard code")
-		}
 	} else if signType == vm.CHECKMULTISIG {
 		// check code and signature
 		if err := checkMultiSigSignatures(program.Program{
 			Code:      info.Code,
-			Parameter: parameter,
+			Parameter: info.Signature,
```

Comment:
should use:
buf := new(bytes.Buffer)
buf.WriteByte(byte(len(info.Signature)))
buf.Write(info.Signature)
parameter := buf.Bytes()